index.js

```diff
@@ -0,0 +1,3 @@
+export function print(str) {
+    console.log(str)
+}
```

Comment:
Sidenote, with JS code we can skip the declaration files, and write with JSDoc comments. So another possibility, and maybe better because TypeScript will actually check that the comment line up with the code, is:

```js
/**
 * Prints the given string `str`.
 * @param {string} str - The string to print
 * @returns {void}
 */
export function print(str) {
    console.log(str)
} 
```

Then if, for example, you use VS Code's refactoring to rename `str` to `theString`, it would modify both the comment and the function.

Reply:
I've actually been wanting to write code like this, because then it means we can write documentation, while avoiding any build steps, and still get type checking.

I was looking if it is possible for AssemblyScript, and there is an outdated project (forget link off top) that converts JSDoc into AssemblyScript-compatible TS (because `asc` doesn't read JSDoc).